src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -1685,7 +1685,10 @@ struct WindowFunctionExponentialTimeDecayedSum final : public StatefulWindowFunc
         auto & state = getState(workspace);
 
         Float64 result = 0;
-        Float64 curr_t = WindowFunctionHelpers::getValue<Float64>(transform, function_index, ARGUMENT_TIME, transform->current_row);
+
+        RowNumber frame_back = transform->frame_end;
+        transform->retreatRowNumber(frame_back);
+        Float64 back_t = WindowFunctionHelpers::getValue<Float64>(transform, function_index, ARGUMENT_TIME, frame_back);
```

Comment:
Let's move it into some helper function like `getValueForFrameEnd` and reuse

Reply:
Probably, but it is too early.

There is pretty strange thing. When we make loop `RowNumber i = transform->frame_start; i < transform->frame_end; transform->advanceRowNumber(i)` it works fine but if we access the supposedly same last value:
```
        RowNumber frame_back = transform->frame_end;
        transform->retreatRowNumber(frame_back);
```

it crashes on:

```
[cristal] 2022.07.29 12:11:04.405001 [ 1185625 ] <Fatal> BaseDaemon: (version 22.8.1.1, build id: 3279EDF860C5A5C3A566CC364A74F40CB89615EC) (from thread 1185605) (query_id: 7d2a97e0-65b4-4664-8d45-caf695126c75) (query: SELECT NULL, exponentialTimeDecayedSum(100000002004087730000.)(value, time) OVER (Rows BETWEEN 255 FOLLOWING AND UNBOUNDED FOLLOWING), number = -2147483649, exponentialTimeDecayedSum(-1)(value, time) OVER (Rows BETWEEN UNBOUNDED PRECEDING AND 65537 PRECEDING), number AS value, number = NULL, exponentialTimeDecayedSum(65537)(value, time) OVER (Rows BETWEEN UNBOUNDED PRECEDING AND 1048577 FOLLOWING), number AS time FROM numbers(65535)) Received signal Segmentation fault (11)
[cristal] 2022.07.29 12:11:04.405049 [ 1185625 ] <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Unknown si_code.
[cristal] 2022.07.29 12:11:04.405077 [ 1185625 ] <Fatal> BaseDaemon: Stack trace: 0x156c8bd1 0x156bcb9c 0x156bd117 0x156c05df 0x154ea883 0x154e19e0 0x154e131d 0x154e0f18 0x154ed3c7 0xb13498e 0xb136cae 0x7f87a6684590 0x7f87a65aa223
[cristal] 2022.07.29 12:11:04.405123 [ 1185625 ] <Fatal> BaseDaemon: 3. DB::WindowFunctionExponentialTimeDecayedSum::windowInsertResultInto(DB::WindowTransform const*, unsigned long) @ 0x156c8bd1 in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405151 [ 1185625 ] <Fatal> BaseDaemon: 4. DB::WindowTransform::writeOutCurrentRow() @ 0x156bcb9c in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405172 [ 1185625 ] <Fatal> BaseDaemon: 5. DB::WindowTransform::appendChunk(DB::Chunk&) @ 0x156bd117 in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405199 [ 1185625 ] <Fatal> BaseDaemon: 6. DB::WindowTransform::work() @ 0x156c05df in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405227 [ 1185625 ] <Fatal> BaseDaemon: 7. DB::ExecutionThreadContext::executeTask() @ 0x154ea883 in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405255 [ 1185625 ] <Fatal> BaseDaemon: 8. DB::PipelineExecutor::executeStepImpl(unsigned long, std::__1::atomic<bool>*) @ 0x154e19e0 in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405279 [ 1185625 ] <Fatal> BaseDaemon: 9. DB::PipelineExecutor::executeImpl(unsigned long) @ 0x154e131d in /home/vladimir/ClickHouseA-bin/programs/clickhouse
[cristal] 2022.07.29 12:11:04.405299 [ 1185625 ] <Fatal> BaseDaemon: 10. DB::PipelineExecutor::execute(unsigned long) @ 0x154e0f18 in /home/vladimir/ClickHouseA-bin/programs/clickhouse
```

Reply:
This however works well:
```
        for (RowNumber i = transform->frame_start; i < transform->frame_end; transform->advanceRowNumber(i))
        {
            back_t = WindowFunctionHelpers::getValue<Float64>(transform, function_index, ARGUMENT_TIME, i);
        }
```

Reply:
Even this does not work, which is extra wierd:
```
        RowNumber frame_back = transform->frame_start;
        //transform->retreatRowNumber(frame_back);
        Float64 back_t = WindowFunctionHelpers::getValue<Float64>(transform, function_index, ARGUMENT_TIME, frame_back);
```

Reply:
@vdimir Is that right that I can reuse `transform->prev_frame_start` to replace my own `previous_frame_start`? And same for `prev_frame_end`?

Reply:
Looks like there is some bug in `WindowTransform`, sometimes window becomes empty `[0, 0)`.

Reply:
Nope, it's fine, user can select empty window.